.spectral.yml

```diff
@@ -0,0 +1,33 @@
+extends: spectral:oas
+rules:
+  oas3-unused-components-schema: off
+  tag-description: off
+  schema-names-pascal-case:
+    description: Schema names MUST be written in PascalCase
+    message: '{{property}} is niet PascalCase: {{error}}'
+    recommended: true
+    type: style
+    given: '$.components.schemas.*~'
+    then:
+      function: pattern
+      functionOptions:
+        match: '^[A-Z][a-zA-Z0-9]*$'
+  property-of-object-type:
+    description: Properties MUST not be of object type
+    message: '{{path}} is gedefinieerd als een inline object. Definieer deze als een schema component en verwijs hiernaar met $ref'
+    type: style
+    resolved: false
+    given: '$.components.schemas..properties[?(@.type === "object")]'
```

Comment:
dit werkt alleen wanneer ook echt type: "object is opgenomen. Dit is default dus kan je ook weglaten.

Als ik dus in common.yaml type: object weghaal wordt geen Warning meer gegeven, bijvoorbeeld bij property first:
```
    HalPaginationLinks:
      allOf:
      - $ref: "#/components/schemas/HalCollectionLinks"
      - type: "object"
        properties:
          first:
            description: "uri voor het opvragen van de eerste pagina van deze collectie"
            properties:
              href:
                type: "string"
                example: "/resourcenaam?page=1"
              templated:
                type: boolean
              title:
                type: "string"
                example: "Eerste pagina"
```

Reply:
Kan je deze niet beter definiëren met:
```
    given: '$.components.schemas..properties[*].properties'
    then:
      function: falsy
```